### PR TITLE
invalidate queries when mutation is settled

### PR DESCRIPTION
**Pull request by joelhooks**
had to use the query client because tRPC invalidate wasn't causing a rerender 🤔

also update react-query and trpc

invalid

### CODE REVIEW

File: apps/total-typescript/src/templates/exercise-template.tsx [hunk 1]
```diff
@@ -50,9 +51,7 @@ const ExerciseTemplate: React.FC<{
           {lessonSlug: router.query.lesson as string},
           {
             onSettled: () => {
-              utils.moduleProgress.bySlug.invalidate({
-                slug: module.slug.current,
-              })
+              client.invalidateQueries()
```
Comment (joelhooks) [inline apps/total-typescript/src/templates/exercise-template.tsx:54]: invalidation via the query client functions as expected and re-renders the with updated progress across the page

File: apps/total-typescript/src/trpc/trpc.client.ts [hunk 2]
```diff
@@ -8,6 +8,24 @@ import type {AppRouter} from 'trpc/routers/_app'
...
+         * The order here allows route changes in `onSuccess` without
+         * having a flash of content change whilst redirecting.
+         **/
+        // Calls the `onSuccess` defined in the `useQuery()`-options:
+        await opts.originalFn()
+        // Invalidate all queries in the react-query cache:
+        await opts.queryClient.invalidateQueries()
+      },
+    },
+  },
```
Comment (joelhooks) [inline apps/total-typescript/src/trpc/trpc.client.ts:28]: EVERY mutation invalidates the cache and reloads

File: apps/total-typescript/src/templates/exercise-template.tsx [hunk 3]
```diff
@@ -38,24 +38,14 @@ const ExerciseTemplate: React.FC<{
...
-        addProgressMutation.mutate(
-          {lessonSlug: router.query.lesson as string},
-          {
-            onSettled: () => {
-              utils.moduleProgress.bySlug.invalidate({
-                slug: module.slug.current,
-              })
-            },
-          },
-        )
```
Comment (joelhooks) [inline apps/total-typescript/src/templates/exercise-template.tsx:58]: we can remove all of these

**Review by vojtaholik — APPROVED** [review]
works great 👍

**Pull request merged**